### PR TITLE
Scalar: imitate .NET-based Scalar's `scalar config` more closely

### PR DESCRIPTION
**Pull request by dscho**
This aligns the order and the categorization into required/optional groups with what is in .NET-based Scalar.

### CODE REVIEW

File: contrib/scalar/scalar.c [hunk 1]
```diff
@@ -178,6 +196,27 @@ static int set_recommended_config(int reconfigure)
...
+	/*
+	 * The `log.excludeDecoration` setting is special because we want to
+	 * set multiple values.
+	 */
+	if (get_config_get_string("log.excludeDecoration", &value)) {
+		trace2_data_string("scalar", the_repository,
+				   "log.excludeDecoration", "created");
+		if (git_config_set_multivar_gently("log.excludeDecoration",
+						   "refs/scalar/*",
+						   CONFIG_REGEX_NONE, 0) ||
```
Comment (derrickstolee) [inline contrib/scalar/scalar.c:208]: This one is leftover from a previous fetch step. Perhaps make a note that we don't need to send this one upstream? Or, maybe we should, anyway. One way to (in a future change) make this optional would be to see if any `refs/scalar/*` ref exists. Don't worry about it now.

**Review by derrickstolee — APPROVED** [review]

File: contrib/scalar/scalar.c (hunk 1 above)
Comment (dscho) [reply contrib/scalar/scalar.c:208]: Okay, I won't, then ;-)

File: contrib/scalar/scalar.c [hunk 2]
```diff
@@ -178,6 +196,27 @@ static int set_recommended_config(int reconfigure)
 		}
 	}
 
+	/*
+	 * The `log.excludeDecoration` setting is special because we want to
+	 * set multiple values.
+	 */
+	if (get_config_get_string("log.excludeDecoration", &value)) {
```
Comment (derrickstolee) [inline contrib/scalar/scalar.c]:
```suggestion
	if (git_config_get_string("log.excludeDecoration", &value)) {
```

Suggested by the broken build.

Comment (dscho) [reply contrib/scalar/scalar.c]: Right, I am on it, locally, but I see a test failure and am investigating before I will push.

**Review by derrickstolee — APPROVED** [review]

**Pull request merged**